### ISSUE TITLE
Improve cardano sign transaction UX

### ISSUE THREAD

**Issue by refi93** [vacuumlabs/trezor-core#1]
The current flow is as follows:

1. confirm fee, tx size
2. confirm network
3. confirm outputs -> confirm address + confirm amount
4. confirm change -> confirm derivation path + confirm amount
5. send tx

Desired flow should mimick bitcoin txs as much as possible
1. for each non-change output print: "confirm sending ${amount} ADA to ${address}"
2. confirm transaction "total amount ${total}, including fee ${fee}, tx size ${txSize} bytes, network ${network}"  -> hold to confirm
3. send tx

inspired by "Bitcoin" sign tx flow: https://www.youtube.com/watch?v=4gSinN8hdeQ&feature=youtu.be&t=1227

conditions for a change address for the purpose of confirming transaction: 
- it must be a derivation path (of form 44'/1815'/x/y/z for the purpose of the conditions below)
- the account prefix (x) must match the account prefix of all the input addresses
- the "change address" portion (y) of the derivation path can be either 1 or 0 (it should be 1 only in theory, but we do this relaxation for AdaLite which uses external addresses for that purpose)
- the "address index" portion (z) must be smaller that 1 000 000 (small enough to brutforce in case of a man-in-the-middle attack)
basically that should be the logic in: https://github.com/vacuumlabs/trezor-core/blob/master/src/apps/wallet/sign_tx/signing.py - if you find some inconsistence between the logic there and the steps above, let me know, most likely it's only my misinterpretation of the code

**Comment by ddeath** [comment]
Hi guys. Just wanted point out that there is similar issue in trezors repository which is kind of addressing it and also the suggested flow by Refi could change in trezor - take a look at the issue:
https://github.com/trezor/trezor-core/issues/380

**Comment by Siegrift** [comment]
@refi93 What approach should I take? Imo people from Trezor need to decide on a UI standard and only then other cryptocurrencies should follow the standard. 

The last commend on [trezor#380](https://github.com/trezor/trezor-core/issues/380) was 14 days ago... Should I reply to the issue, that we would like to fix the Cardano UI?

**Comment by refi93** [comment]
@ddeath thanks for pointing out

@Siegrift I think you can start with implementing the path check for change addresses, so we don't display it when it's safe to, btw it's summed up nicely here for bitcoin: https://github.com/trezor/trezor-core/blob/master/docs/coins/bitcoin-path-check.svg which I'm quite confident that would be needed anyway.

I already asked them in the thread, so hopefully they reply soon and we will proceed accordingly.

**Comment by refi93** [comment]
Btw @ddeath , do you think the proposed flow is feasible? Do you see any problem with pagination/amounts?

**Comment by ddeath** [comment]
The paginating could be problem for large data. It is more an edge case but maybe you noticed that the cardano addresses could be (i think 64KB) long. As ppershing discovered some guy already played with it and encoded and image in it :smile: and it is still valid cardano address... The point is in this case you would get "infinite" pagination.... But as I said it is an edge case so probably would not bother with it...

**Comment by refi93** [comment]
Would such a tx even be possible to send to Trezor, isn't the limit 4kB? But yes, for such a rare case, we could possibly add some ellipsis on the address, showing only 256 characters or so :D
